frontend/lib/src/components/core/Block/styled-components.ts

```diff
@@ -130,6 +131,19 @@ export const StyledColumn = styled.div<StyledColumnProps>(
       [`@media (max-width: ${theme.breakpoints.columns})`]: {
         minWidth: `calc(100% - ${theme.spacing.twoXL})`,
       },
+      ...(verticalAlignment === "bottom" && {
+        marginTop: "auto",
+        // Add margin to the last checkbox within the column to align it
+        // better with other input widgets. This is a temporary (ugly) fix
+        // until we have a better solution for this.
+        "& .element-container:last-of-type > .stCheckbox": {
```

Comment:
I don't understand this, can I see an example with images or something.

I don't prefer to target classes in favor of referencing Emotion components. Perhaps we should discuss the problem and find a better solution.

Reply:
It adding a small margin to the last element in a column if its a checkbox. 

Without this fix:

<img width="653" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/82a241ea-4090-46d5-9fe3-2f26acc42764">

With the fix (we want the checkbox to be centered with the widget input fields): 

<img width="653" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/49a053e5-d1cd-4369-9908-83d5d4012655">